### DISCUSSION TITLE
Quasar (Version 1) and vue-echarts

### DISCUSSION THREAD

**Discussion by susansiow** [quasarframework/quasar#10678] [General - Components / Directives / etc] [answered]
Dear Quasar,

I've been searching around for the info about incorporating vue-echarts with Quasar. As described in its [documentation](https://github.com/ecomfe/vue-echarts), in the Vue2 environment, I need to install Vue Composition-API (which I've complied with it) and secondly, to register below in main.js (which I was not able to fulfill it as I believe it needs to go through Quasar boot file). 

Kindly advise how I can proceed with the second requirement in Quasar (Version 1.14.5)?

```
import Vue from 'vue'
import ECharts from 'vue-echarts'
import { use } from 'echarts/core'

// import ECharts modules manually to reduce bundle size
import {
  CanvasRenderer
} from 'echarts/renderers'
import {
  BarChart
} from 'echarts/charts'
import {
  GridComponent,
  TooltipComponent
} from 'echarts/components'

use([
  CanvasRenderer,
  BarChart,
  GridComponent,
  TooltipComponent
]);

// register globally (or you can do it locally)
Vue.component('v-chart', ECharts)
```

**Comment by metalsadman** [comment]
just copy paste your code above in a boot file, then register it in your `quasar.conf.js>boot` array. https://v1.quasar.dev/quasar-cli/boot-files#introduction

**Reply by susansiow** [reply]
I've created a file named 'echarts.js' under the 'boot' folder with the following code:

```
import Vue from 'vue'
import ECharts from 'vue-echarts'
import { use } from 'echarts/core'

// import ECharts modules manually to reduce bundle size
import {
  CanvasRenderer
} from 'echarts/renderers'
import {
  BarChart
} from 'echarts/charts'
import {
  GridComponent,
  TooltipComponent
} from 'echarts/components'

use([
  CanvasRenderer,
  BarChart,
  GridComponent,
  TooltipComponent
])

export default async ({ app }) => {
    app.component('v-chart', ECharts)
}
```

And in quasar.conf.js I've added 'echarts' (the file name) in the boot array:

```
boot: [
      'axios',
      'echarts'
],
```

But when I run 'quasar dev', I've received this error in the console.

<img width="431" alt="s" src="https://user-images.githubusercontent.com/53169547/133381097-257376d8-79bf-45c3-ba3c-93667bbf0382.png">

What did I miss out?

**Comment by MilosPaunovic** [comment]
You are importing the library in a Vue 3 way, and say that you use Quasar 1 (with Vue 2.x). 

There is a section on [offical documentation](https://www.npmjs.com/package/vue-echarts) about importing in the Vue 2 way, you just have to expand it to see it.

**Reply by susansiow** [reply]
I have double checked the official documentation, in fact, I'm applying the Vue 2 way, below is the screenshot.

<img width="330" alt="s" src="https://user-images.githubusercontent.com/53169547/133390001-cc1f2284-33f8-463f-8e94-e94fcf2ebc2b.png">

Please help, thank you!

**Reply by pratik227** [reply]
@susansiow You are referring wrong npm link so latest version link is for Vue 3

**Comment by ontwikkelfabriek** [comment, marked as answer]
@susansiow 

This Quasar v1 app uses the plain JS version of echarts, without the **forced** use of composition API and boot files configs:
https://github.com/pratik227/quasar-echarts

**Reply by susansiow** [reply]
@ontwikkelfabriek 

The reason I'm applying vue-echarts (v6) with Quasar (v1) because I need to show the chart demo to the client before upgrading the app to Quasar (v2). After upgrading to Quasar (v2), the app cannot support vue-echarts (before v6), correct? vue-echarts-v3 is so much easier to configure but the Quasar upgrading matter and the 'looks-like-going-to-be-outdated' [vue-echarts-v3](https://www.npmjs.com/package/vue-echarts-v3) (last published date: 3 years ago) are the reasons of why I'm trying to setup vue-echarts (v6) instead.

Hope to hear your thoughts on this, thank you!

**Reply by ontwikkelfabriek** [reply]
@susansiow 

No. That's not what I am suggesting. I offer an alternative, instead of using **vue**-echart-v3/v4/v6, the app in my posted repo uses the **non** vue, vanilla echart.js library. This way it doesn't matter if you are using Qv1 or Qv2 or Qv3...

**Reply by susansiow** [reply]
@ontwikkelfabriek 

I get it and it works, thank you so much!!!

Lastly, just want to gain enlightenment on whether the integration of cdnjs will cause security issue for a private app (not public app/app which requires logged in authentication)?
